flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/bufferdebloat/BufferDebloater.java

```diff
@@ -96,7 +96,9 @@ public void recalculateBufferSize(long currentThroughput) {
 
         lastBufferSize = newSize;
         for (IndexedInputGate inputGate : inputGates) {
-            inputGate.announceBufferSize(newSize);
+            if (!inputGate.isFinished()) {
```

Comment:
What about `released`? Should we somehow check for that as well?

Reply:
We check for `released`(for channels) inside of `inputGate#announceBufferSize` which is enough for this backport.